webui/react/src/hooks/useSettings.ts

```diff
@@ -142,35 +142,22 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
   });
   const { isLoading, querySettings, state, update } = useContext(UserSettings);
   const navigate = useNavigate();
-  const pathname = window.location.pathname;
+  const location = useLocation();
+  const pathname = location.pathname;
   const shouldSkipUpdates = useMemo(
     () => config.applicableRoutespace && !pathname.endsWith(config.applicableRoutespace),
     [config.applicableRoutespace, pathname],
   );
-
-  // parse navigation url to state
-  useEffect(() => {
-    if (!querySettings || shouldSkipUpdates) return;
-
-    const settings = queryToSettings<T>(config, querySettings);
-    const stateSettings = state.get(config.storagePath) ?? {};
-
-    if (isEqual(settings, stateSettings)) return;
-
-    Object.keys(settings).forEach((setting) => {
-      stateSettings[setting] = settings[setting];
-    });
-
-    update(config.storagePath, stateSettings, true);
-  }, [config, querySettings, state, update, shouldSkipUpdates]);
+  const [shouldPush, setShouldPush] = useState(false);
```

Comment:
nit: `useState<boolean>(false);` is more preferable. 
and nice to have the comment of purpose of this variable